algorithms/generated/scc.gsql

```diff
@@ -0,0 +1,174 @@
+                        curr = active;
+                        WHILE (curr.size()>0) DO
+                                curr = SELECT t
+                                       FROM curr:s -((link_to|link_to2|link_to3|reverse_link_to|reverse_link_to2|reverse_link_to3):e)-> :t
+                                       WHERE s.@cid == t.@cid AND t.@detached == FALSE
+                                       ACCUM t.@wcc_id_curr += s.@wcc_id_prev // If s has a smaller id than t, copy the id to t
+                                       POST-ACCUM
+                                                CASE WHEN t.@wcc_id_prev != t.@wcc_id_curr THEN // If t's id has changed
+                                                          t.@wcc_id_prev = t.@wcc_id_curr,
+                                                          t.@wcc_active = true
+                                                ELSE 
+                                                          t.@wcc_active = false
+                                                END
+                                        HAVING t.@wcc_active == true;       
+                        END;
+                        active = SELECT s
+                                 FROM active:s
+                                 ACCUM s.@cid = s.@wcc_id_curr;
+                END;
+                i = i + 1;
+  
+                pivots = SELECT s
+                         FROM active:s 
+                         WHERE s.@cid == s.@vid
+                         ACCUM s.@is_forward = TRUE,
+                               s.@is_backward = TRUE;
+          
+                // mark forward set
+                curr = pivots;
+                WHILE curr.size()>0 DO
+                        curr = SELECT t 
+                               FROM curr:s -((link_to|link_to2|link_to3):e)->:t  // edge
+                               WHERE t.@detached == FALSE AND t.@is_forward == FALSE AND s.@cid == t.@cid // not traversed
+                               ACCUM t.@is_forward = TRUE;
+                END;
+          
+                // mark backward set
+                curr = pivots;
+                WHILE curr.size()>0 DO
+                        curr = SELECT t 
+                               FROM curr:s -((reverse_link_to|reverse_link_to2|reverse_link_to3):e)->:t  // reverse edge
+                               WHERE t.@detached == FALSE AND t.@is_backward == FALSE AND s.@cid == t.@cid // not traversed
+                               ACCUM t.@is_backward = TRUE;
+                END;
+          
+                active = SELECT s
+                         FROM active:s 
+                         ACCUM IF s.@is_forward == TRUE AND s.@is_backward == TRUE THEN  // scc
+                                     s.@detached = TRUE,
+                                     @@s_cid_map += (s.@cid -> s.@vid)
+                               END,
+                               IF s.@is_forward == TRUE THEN
```

Comment:
maybe we could simplify the if else structure?

if f == true and b == ture then
else if f == true
else if b == ture
end